GATEWAY/src/providers/devices.service.ts

```diff
@@ -19,6 +19,9 @@ export class DevicesService {
   public getDevices = (): Device[] => {
     return this.devices;
   }
+  public setDevices = (dev: Device[]): void => {
```

Comment:
to set new devices call newdevice several times instead. This method can be confusing for the developers who are taking over the project as it should never be used by non-testing code.